gluten-core/src/main/scala/org/apache/gluten/extension/columnar/transition/ConventionFunc.scala

```diff
@@ -92,6 +92,12 @@ object ConventionFunc {
         }
         val conv = Convention.of(rowType, batchType)
         conv
+      case _: ColumnarWriteFilesExec.ExecuteWriteCompatible =>
+        // ColumnarWriteFilesExec neither output Row nor columnar data.
+        // We output both row and columnar to avoid c2r and r2c transitions.
+        Convention.of(
+          Convention.RowType.VanillaRow,
+          BackendsApiManager.getSparkPlanExecApiInstance.batchType)
```

Comment:
Basically this list is for overriding conventions of vanilla Spark's operators. Would you use interface `KnowBatchType` / `KnownRowType` instead?